src/App.tsx

```diff
@@ -1,61 +1,132 @@
 import './App.scss';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import React, { FormEvent, useState } from 'react';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import { Todo } from './types/Todo';
+import { TodoList } from './components/TodoList';
+
+export const App: React.FC = () => {
+  const preparedTodos = () => {
+    return todosFromServer.map(todo => {
+      const user = usersFromServer.find(
+        userFromServer => userFromServer.id === todo.userId,
+      );
+
+      return { ...todo, user: user || null };
+    });
+  };
+
+  const [tasks, setTasks] = useState<Todo[]>(preparedTodos());
+
+  const [userId, setUserId] = useState(0);
+  const [title, setTitle] = useState('');
+
+  const [errorUserId, setErrorUserId] = useState('');
+  const [errorTitle, setErrorTitle] = useState('');
+
+  const handleEnteredTitle = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value);
+
+    if (errorTitle) {
+      setErrorTitle('');
+    }
+  };
+
+  const handleUserSelect = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setUserId(+event.target.value);
+
+    if (errorUserId) {
+      setErrorUserId('');
+    }
+  };
+
+  const handleReset = () => {
+    setUserId(0);
+    setTitle('');
+    setErrorTitle('');
+    setErrorUserId('');
+  };
+
+  const handleSubmit = (event: FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (!title || userId === 0) {
+      if (!title) {
+        setErrorTitle('Please enter a title');
+      }
+
+      if (userId === 0) {
+        setErrorUserId('Please choose a user');
+      }
+
+      return;
+    }
+
+    const user = usersFromServer.find(
+      userFromServer => userFromServer.id === userId,
+    );
+
+    const newTaskId =
+      tasks.length > 0 ? Math.max(...tasks.map(task => task.id)) + 1 : 1;
+
+    const newTask: Todo = {
+      id: newTaskId,
+      title: title,
+      userId: userId,
+      completed: false,
+      user: user || null,
+    };
+
+    setTasks(currentTasks => [...currentTasks, newTask]);
+
+    handleReset();
+  };
 
-export const App = () => {
   return (
     <div className="App">
       <h1>Add todo form</h1>
 
-      <form action="/api/todos" method="POST">
+      <form action="/api/todos" method="POST" onSubmit={handleSubmit}>
         <div className="field">
-          <input type="text" data-cy="titleInput" />
-          <span className="error">Please enter a title</span>
+          <span>Title: </span>
```

Comment:
According to task check-list :
*add labels and placeholders where they are needed;

You have to change your span tag by a <label> here and at the line 104
You can also check it if you enter the working version example and try to click on label Title  you will be able to start entering  somethig in the input field.
```suggestion
        <label className="label" htmlFor={'title'}>
          Title:{' '}
        </label>
```